build: boring autotools cleanup

fanquake opened the pull request:
This is mostly just being consistent with how we do things, and migrating towards a style (we have already been doing so ad-hoc) that is clearer for anyone who cares to read `.m4`. For example:

`master`:
```m4
  AX_CHECK_COMPILE_FLAG(
    [-g3],
    [[DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -g3"]],
    [AX_CHECK_COMPILE_FLAG([-g],[[DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -g"]],,[[$CXXFLAG_WERROR]])],
    [[$CXXFLAG_WERROR]])
```

This PR:
```m4
  AX_CHECK_COMPILE_FLAG(
    [-g3],
    [DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -g3"],
    [AX_CHECK_COMPILE_FLAG([-g], [DEBUG_CXXFLAGS="$DEBUG_CXXFLAGS -g"], [], [$CXXFLAG_WERROR])],
    [$CXXFLAG_WERROR])
```

Drop unneeded double-quoting (which we use inconsistently), use `[]` for empty arguments, space things out.

There should be no functional change, before & after binaries identical. Very boring.

Guix build:
```bash
bash-5.1# find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
22097cd621cd88348f827b916f4b4b120b40c3515a3752595347e36d57dc9158  guix-build-34094aff1348/output/aarch64-linux-gnu/SHA256SUMS.part
43f10bb857afa7ea52a8ed9beed36ff0e3ee02dba31100fc04c8e0b2443d94eb  guix-build-34094aff1348/output/aarch64-linux-gnu/bitcoin-34094aff1348-aarch64-linux-gnu-debug.tar.gz
9095a8228376094065103f4adc9cdcb8189111fb9536ad88e4f8cecc3df6fc75  guix-build-34094aff1348/output/aarch64-linux-gnu/bitcoin-34094aff1348-aarch64-linux-gnu.tar.gz
9c73179059c6fe1f7643445ec5a530003fc41187aa0a94cb1f5c106097161e5b  guix-build-34094aff1348/output/arm-linux-gnueabihf/SHA256SUMS.part
70ddd4dd0a06c7491937084125f690c1d62fa2647c16048fc1a4a9a09d8b10b4  guix-build-34094aff1348/output/arm-linux-gnueabihf/bitcoin-34094aff1348-arm-linux-gnueabihf-debug.tar.gz
0a338fdc9788c33a0d519b6c09fdf6271e3bd68846ee61eef0a06a2df6bab419  guix-build-34094aff1348/output/arm-linux-gnueabihf/bitcoin-34094aff1348-arm-linux-gnueabihf.tar.gz
25eda7fae2984b9dadf47420d1dc09b0224d425144233482602bd9e6d348255c  guix-build-34094aff1348/output/dist-archive/bitcoin-34094aff1348.tar.gz
d70d84e43ffa2d809063cda868f708539e9114b2d14edb6ddcf05fdf73f3187b  guix-build-34094aff1348/output/powerpc64-linux-gnu/SHA256SUMS.part
8d1291e576d2b5f8f7120fe6e6ed4b23415249e22a657a350ccce68ff261e088  guix-build-34094aff1348/output/powerpc64-linux-gnu/bitcoin-34094aff1348-powerpc64-linux-gnu-debug.tar.gz
eab448186aee18ac33c39eed4d24501208d10d257fe6e2739adf589b1d4b693a  guix-build-34094aff1348/output/powerpc64-linux-gnu/bitcoin-34094aff1348-powerpc64-linux-gnu.tar.gz
4d28617b4d0ddb88c8b20d06ca21314ee40814043f92cabcd9ea3e3d8ee39183  guix-build-34094aff1348/output/powerpc64le-linux-gnu/SHA256SUMS.part
dea02168e170e92600012f5806ec8b39209282c2270669f2040682f74bc3f320  guix-build-34094aff1348/output/powerpc64le-linux-gnu/bitcoin-34094aff1348-powerpc64le-linux-gnu-debug.tar.gz
c7122e89d3186a183ac08e7f3020654722c98bf8acc8b790bb292b39f5ba8225  guix-build-34094aff1348/output/powerpc64le-linux-gnu/bitcoin-34094aff1348-powerpc64le-linux-gnu.tar.gz
0802a52720d2bec1264dc13f6554a9da347baa3d096242b29bb524f4b121eb10  guix-build-34094aff1348/output/riscv64-linux-gnu/SHA256SUMS.part
4aee9fc41f35d2adb2d0562902dd8584a8413a73c015ddcdcef00586779f63a7  guix-build-34094aff1348/output/riscv64-linux-gnu/bitcoin-34094aff1348-riscv64-linux-gnu-debug.tar.gz
3f0c4d6096ac7e08389e851c2d252632c044a700ce0174473ac4d7f66290e8cc  guix-build-34094aff1348/output/riscv64-linux-gnu/bitcoin-34094aff1348-riscv64-linux-gnu.tar.gz
c5149c46b9b7081d5715daf3e22fd30ffca23d333f664da1fabc8143ff8bf76c  guix-build-34094aff1348/output/x86_64-apple-darwin19/SHA256SUMS.part
81848355751e55a8a60636e3ea2f03ca6abb78736a5431715cd51cebd46bb961  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx-unsigned.dmg
d2f6f689cbedddd865f90dba9ddf21479c71c61b0350fda62804b2f233116a43  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx-unsigned.tar.gz
99d8fe428fcb67f9975e6b8d9a63d84946215a0a6b8f94967ce96cc3af4b7772  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx64.tar.gz
3dac13c7556d9a25ff5513bbb2638fe4fa74d8a88304bbdce52364df7832a3ab  guix-build-34094aff1348/output/x86_64-linux-gnu/SHA256SUMS.part
e4baa7da80fdabbb50953efaaa7b4867c7e575a7a156b728e8e197142df55697  guix-build-34094aff1348/output/x86_64-linux-gnu/bitcoin-34094aff1348-x86_64-linux-gnu-debug.tar.gz
f82f5bcc7197c1741b106f62be7b468aadbdf5b3198091582026cd450bf13b3a  guix-build-34094aff1348/output/x86_64-linux-gnu/bitcoin-34094aff1348-x86_64-linux-gnu.tar.gz
db22b5f48783917f985920ddb26aa170b4d6cc65112406548847a883099505b8  guix-build-34094aff1348/output/x86_64-w64-mingw32/SHA256SUMS.part
ead0809193ca1d462553a6f3f233cdbff7a3f8419100d825abfc10835508e485  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win-unsigned.tar.gz
e1006b6c114eaf33274144d8e9a20abc0cee01e26a4594063ee615bf09c1b344  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64-debug.zip
2d3fbc593b58d353a6859e02c0dd096d453cf5f1e3150c2a1c234bdfc97b4f24  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64-setup-unsigned.exe
25cfa8a3655727572593f100b7a70c1061fa6f3b017014ad7444059611c3ddda  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64.zip
```

fanquake committed:
build: AX_CHECK_LINK_FLAG serial 6

fanquake committed:
build: cleanup AX_CHECK_LINK_FLAG() usage

There should be no functional change.

fanquake committed:
build: AX_CHECK_COMPILE_FLAG() serial 6

fanquake committed:
build: cleanup AX_CHECK_COMPILE_FLAG() usage

There should be no functional change.

fanquake committed:
build: AX_CHECK_PREPROC_FLAG() serial 6

fanquake committed:
build: cleanup AX_CHECK_PREPROC_FLAG() usage

There should be no functional change.

fanquake committed:
build: AC_PATH_PROG(S) consistently quote arguments

fanquake committed:
build: consistently quote AC_PATH_TOOL arguments

fanquake committed:
build: consistently quote AC_MSG_CHECKING() arguments

fanquake committed:
build: consistently quote AC_DEFINE() arguments

fanquake committed:
build: consistently quote arguments in AM_CONDITIONAL()

fanquake committed:
build: consistently quote arguments in AC_ARG_VAR()

hebasto commented:
> Drop unneeded double-quoting (which we use inconsistently), use `[]` for empty arguments, space things out.

Concept ACK, obviously, as I'm using the same style :)

Why update Autoconf macros as they have no functional changes?

katesalazar commented:
Mostly concept ACK. Checked (only) http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_check_preproc_flag.m4 to see licensing changes come from upstream.

fanquake commented:
> Why update Autoconf macros as they have no functional changes?

Mostly just why not, and if we do it now, it saves someone else wanting to do it in the future. I'd think that these files will be very unlikely to be ever updated again after this.

hebasto reviewed (commented):
Suggesting to add more boring stuff:
- `AC_CHECK_PROG`
- `AC_MSG_WARN`
- `AC_MSG_ERROR`
- `AC_MSG_RESULT`

Why only the first and second arguments are quoted in `AC_PATH_PROG(S)` and `AC_PATH_TOOL`?

jamesob commented:
Concept ACK, will at least check for binary identity soon.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
9e437118c7dc07635517372fda7abc7b64032de631e9c46d6cdd0f61337b7267  guix-build-9f0a31662420/output/aarch64-linux-gnu/SHA256SUMS.part
d2d2650953f7fb29e49191cccb441351108f9eaabd540c230d19c27879a1f739  guix-build-9f0a31662420/output/aarch64-linux-gnu/bitcoin-9f0a31662420-aarch64-linux-gnu-debug.tar.gz
fb8d70ca94abd0963227c127c4a907bcf21a3339586d6198306240b9f43d55e7  guix-build-9f0a31662420/output/aarch64-linux-gnu/bitcoin-9f0a31662420-aarch64-linux-gnu.tar.gz
e20131efb154db2b106586c73b97480ba2199684c6f4b3067a72b69d512d9d8c  guix-build-9f0a31662420/output/arm-linux-gnueabihf/SHA256SUMS.part
da101fa188f0a5b72c6a3542cfd41d372a0295369659b46988a7d5bb794cbdad  guix-build-9f0a31662420/output/arm-linux-gnueabihf/bitcoin-9f0a31662420-arm-linux-gnueabihf-debug.tar.gz
36b8d8c1b6cd68228e771edd2d871223b0c57c523a1aeb33d7fcfa6b56e2fa5d  guix-build-9f0a31662420/output/arm-linux-gnueabihf/bitcoin-9f0a31662420-arm-linux-gnueabihf.tar.gz
672dcc4463f151fdcc4d0a4c45d846640998939eba2a7e31582bd23b334d379f  guix-build-9f0a31662420/output/dist-archive/bitcoin-9f0a31662420.tar.gz
3843aa4ef36f235c47cbcd2f272ccb1e5d4146dfd11d147fa3743ab30e9bb408  guix-build-9f0a31662420/output/powerpc64-linux-gnu/SHA256SUMS.part
6385015180b123c760784cec0aef76d225f88b214d8a069a950812dbbecf072c  guix-build-9f0a31662420/output/powerpc64-linux-gnu/bitcoin-9f0a31662420-powerpc64-linux-gnu-debug.tar.gz
7a3cab441850deeb9ffbef8450d0b643216d5247eb4b2496622d744333ab87f6  guix-build-9f0a31662420/output/powerpc64-linux-gnu/bitcoin-9f0a31662420-powerpc64-linux-gnu.tar.gz
9bbb0fc21da13e50a7477026fcdfa5a42067096132138b24ca81cd65f8bc215c  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/SHA256SUMS.part
b7f9f506020eda38c9454ab524fe5e1d0fbe948bcc52bf9d2e04583d54186e37  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/bitcoin-9f0a31662420-powerpc64le-linux-gnu-debug.tar.gz
14698193c2539e57ed35c0d755d90dec6e90e1bdebb02ee93420c0f30df58c06  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/bitcoin-9f0a31662420-powerpc64le-linux-gnu.tar.gz
aa87c72a8bc79d50e2d9314e067af62317c5feebab95eb6a5aef910caf988c8d  guix-build-9f0a31662420/output/riscv64-linux-gnu/SHA256SUMS.part
bcee4d7280a4056ac8c425fdc401b412e31ffb69b392476d4c0619b8634ecf93  guix-build-9f0a31662420/output/riscv64-linux-gnu/bitcoin-9f0a31662420-riscv64-linux-gnu-debug.tar.gz
0aecc7f729db5856019ebcec14c5e9fd9cdc7556eef45797b28e1aebe2cf52bd  guix-build-9f0a31662420/output/riscv64-linux-gnu/bitcoin-9f0a31662420-riscv64-linux-gnu.tar.gz
4c4fa135ce22f8da63e79c9a8c4c4a39d8e491f9d0738c631c864665cb46c14f  guix-build-9f0a31662420/output/x86_64-apple-darwin19/SHA256SUMS.part
b82fbc5166013e3d3deceb6d6a6f0836f79686ab1a502dad5c6da946bd6bcee9  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx-unsigned.dmg
f3ee3de8963bfc5395d03c074e7af144ea54b9328c4bfb2d4edee58f9f71f26e  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx-unsigned.tar.gz
6ae04d3ea227728cfca4234940d3f27ba0afa519f16ad3f5506330c6c93497f3  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx64.tar.gz
0fac8e1ee8cff7f68cfda467741ddac101d4cd7e1d80824a8866d23710f40bc4  guix-build-9f0a31662420/output/x86_64-linux-gnu/SHA256SUMS.part
29615e367b2bdd3f5cc03ac1ce9529b862040761615e0bc46b48f18e2ca39c6a  guix-build-9f0a31662420/output/x86_64-linux-gnu/bitcoin-9f0a31662420-x86_64-linux-gnu-debug.tar.gz
c2df5db5be077235bc265f97e1fe018cfd042f8b60e67a9bd4344f91aa23a166  guix-build-9f0a31662420/output/x86_64-linux-gnu/bitcoin-9f0a31662420-x86_64-linux-gnu.tar.gz
a0fcc8c9a22c9ce70cae0095af259bbf80f1a1ce7c0618cf085f64c01e042e72  guix-build-9f0a31662420/output/x86_64-w64-mingw32/SHA256SUMS.part
732588fa6a4972b268148579ff795a534a96c7973edc5db103acea59ce0a3fc9  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win-unsigned.tar.gz
1d06cea5c7a4e0d1d01d6f8a744d4d517f7961b0d56ae2438fcff8825d365d77  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64-debug.zip
8f6f1bfc6c94097b0f1ce01a5f40bc0ae07f3b6e31daa9c8586cac4e906dc52f  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64-setup-unsigned.exe
9e1d2a3ada18b3542e079cb252b10453414b34ce0e70aee27997da51055a14cc  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64.zip
```

jarolrod commented:
GUIX hashes, mine match @hebasto and @fanquake

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

9e437118c7dc07635517372fda7abc7b64032de631e9c46d6cdd0f61337b7267  guix-build-9f0a31662420/output/aarch64-linux-gnu/SHA256SUMS.part
d2d2650953f7fb29e49191cccb441351108f9eaabd540c230d19c27879a1f739  guix-build-9f0a31662420/output/aarch64-linux-gnu/bitcoin-9f0a31662420-aarch64-linux-gnu-debug.tar.gz
fb8d70ca94abd0963227c127c4a907bcf21a3339586d6198306240b9f43d55e7  guix-build-9f0a31662420/output/aarch64-linux-gnu/bitcoin-9f0a31662420-aarch64-linux-gnu.tar.gz
e20131efb154db2b106586c73b97480ba2199684c6f4b3067a72b69d512d9d8c  guix-build-9f0a31662420/output/arm-linux-gnueabihf/SHA256SUMS.part
da101fa188f0a5b72c6a3542cfd41d372a0295369659b46988a7d5bb794cbdad  guix-build-9f0a31662420/output/arm-linux-gnueabihf/bitcoin-9f0a31662420-arm-linux-gnueabihf-debug.tar.gz
36b8d8c1b6cd68228e771edd2d871223b0c57c523a1aeb33d7fcfa6b56e2fa5d  guix-build-9f0a31662420/output/arm-linux-gnueabihf/bitcoin-9f0a31662420-arm-linux-gnueabihf.tar.gz
672dcc4463f151fdcc4d0a4c45d846640998939eba2a7e31582bd23b334d379f  guix-build-9f0a31662420/output/dist-archive/bitcoin-9f0a31662420.tar.gz
3843aa4ef36f235c47cbcd2f272ccb1e5d4146dfd11d147fa3743ab30e9bb408  guix-build-9f0a31662420/output/powerpc64-linux-gnu/SHA256SUMS.part
6385015180b123c760784cec0aef76d225f88b214d8a069a950812dbbecf072c  guix-build-9f0a31662420/output/powerpc64-linux-gnu/bitcoin-9f0a31662420-powerpc64-linux-gnu-debug.tar.gz
7a3cab441850deeb9ffbef8450d0b643216d5247eb4b2496622d744333ab87f6  guix-build-9f0a31662420/output/powerpc64-linux-gnu/bitcoin-9f0a31662420-powerpc64-linux-gnu.tar.gz
9bbb0fc21da13e50a7477026fcdfa5a42067096132138b24ca81cd65f8bc215c  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/SHA256SUMS.part
b7f9f506020eda38c9454ab524fe5e1d0fbe948bcc52bf9d2e04583d54186e37  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/bitcoin-9f0a31662420-powerpc64le-linux-gnu-debug.tar.gz
14698193c2539e57ed35c0d755d90dec6e90e1bdebb02ee93420c0f30df58c06  guix-build-9f0a31662420/output/powerpc64le-linux-gnu/bitcoin-9f0a31662420-powerpc64le-linux-gnu.tar.gz
aa87c72a8bc79d50e2d9314e067af62317c5feebab95eb6a5aef910caf988c8d  guix-build-9f0a31662420/output/riscv64-linux-gnu/SHA256SUMS.part
bcee4d7280a4056ac8c425fdc401b412e31ffb69b392476d4c0619b8634ecf93  guix-build-9f0a31662420/output/riscv64-linux-gnu/bitcoin-9f0a31662420-riscv64-linux-gnu-debug.tar.gz
0aecc7f729db5856019ebcec14c5e9fd9cdc7556eef45797b28e1aebe2cf52bd  guix-build-9f0a31662420/output/riscv64-linux-gnu/bitcoin-9f0a31662420-riscv64-linux-gnu.tar.gz
4c4fa135ce22f8da63e79c9a8c4c4a39d8e491f9d0738c631c864665cb46c14f  guix-build-9f0a31662420/output/x86_64-apple-darwin19/SHA256SUMS.part
b82fbc5166013e3d3deceb6d6a6f0836f79686ab1a502dad5c6da946bd6bcee9  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx-unsigned.dmg
f3ee3de8963bfc5395d03c074e7af144ea54b9328c4bfb2d4edee58f9f71f26e  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx-unsigned.tar.gz
6ae04d3ea227728cfca4234940d3f27ba0afa519f16ad3f5506330c6c93497f3  guix-build-9f0a31662420/output/x86_64-apple-darwin19/bitcoin-9f0a31662420-osx64.tar.gz
0fac8e1ee8cff7f68cfda467741ddac101d4cd7e1d80824a8866d23710f40bc4  guix-build-9f0a31662420/output/x86_64-linux-gnu/SHA256SUMS.part
29615e367b2bdd3f5cc03ac1ce9529b862040761615e0bc46b48f18e2ca39c6a  guix-build-9f0a31662420/output/x86_64-linux-gnu/bitcoin-9f0a31662420-x86_64-linux-gnu-debug.tar.gz
c2df5db5be077235bc265f97e1fe018cfd042f8b60e67a9bd4344f91aa23a166  guix-build-9f0a31662420/output/x86_64-linux-gnu/bitcoin-9f0a31662420-x86_64-linux-gnu.tar.gz
a0fcc8c9a22c9ce70cae0095af259bbf80f1a1ce7c0618cf085f64c01e042e72  guix-build-9f0a31662420/output/x86_64-w64-mingw32/SHA256SUMS.part
732588fa6a4972b268148579ff795a534a96c7973edc5db103acea59ce0a3fc9  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win-unsigned.tar.gz
1d06cea5c7a4e0d1d01d6f8a744d4d517f7961b0d56ae2438fcff8825d365d77  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64-debug.zip
8f6f1bfc6c94097b0f1ce01a5f40bc0ae07f3b6e31daa9c8586cac4e906dc52f  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64-setup-unsigned.exe
9e1d2a3ada18b3542e079cb252b10453414b34ce0e70aee27997da51055a14cc  guix-build-9f0a31662420/output/x86_64-w64-mingw32/bitcoin-9f0a31662420-win64.zip
```

fanquake committed:
build: consistently quote AC_CHECK_PROG() arguments

fanquake committed:
build: consistently quote AC_MSG_* arguments

fanquake committed:
build: consistently quote AC_CHECK_LIB() arguments

fanquake commented:
> Why only the first and second arguments are quoted in AC_PATH_PROG(S) and AC_PATH_TOOL?

Updated.

> Suggesting to add more boring stuff:

Added.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
39f52693e947cdf654aaef888fce45b56fce21b2d71e031f4d1ed8f58bd07530  guix-build-d572dd305094/output/aarch64-linux-gnu/SHA256SUMS.part
b787bc6e3632d0a4c94df710c23d4526cf9b33d256b5cdddc7701655164bd031  guix-build-d572dd305094/output/aarch64-linux-gnu/bitcoin-d572dd305094-aarch64-linux-gnu-debug.tar.gz
4126e892eac7007c8996e55259cd855965ab89cf5b5fd23b8bb51fa2094f3d67  guix-build-d572dd305094/output/aarch64-linux-gnu/bitcoin-d572dd305094-aarch64-linux-gnu.tar.gz
a6590e8f88cb4b3eeccec4bb1e34b471c7d5f45ea856002c9b31c6836bb863f3  guix-build-d572dd305094/output/arm-linux-gnueabihf/SHA256SUMS.part
70658f776bd203fa1d685242590da045dd413c085639642b504b2597b7389b30  guix-build-d572dd305094/output/arm-linux-gnueabihf/bitcoin-d572dd305094-arm-linux-gnueabihf-debug.tar.gz
9eeda21c455711fd1bcc8fb0b1fc7e41962ed9d8d07a428f20c83269da3ab97d  guix-build-d572dd305094/output/arm-linux-gnueabihf/bitcoin-d572dd305094-arm-linux-gnueabihf.tar.gz
e574c238c55d69df78b1fcec2e61010efab14762d66b3cf5ca7492b99d0cdbe9  guix-build-d572dd305094/output/dist-archive/bitcoin-d572dd305094.tar.gz
79e70352d5ee56ca8a29165a852bfcf7492b5909e504752e75c98715b2a9432f  guix-build-d572dd305094/output/powerpc64-linux-gnu/SHA256SUMS.part
f91ca41d6374d8c6833a12cf1aba5daa087f5d9ebc8f671958f2141f22726d87  guix-build-d572dd305094/output/powerpc64-linux-gnu/bitcoin-d572dd305094-powerpc64-linux-gnu-debug.tar.gz
c0a54f61572a008d7c503f249bcb7bb462824e692879ffdd7f82ed68f7259a38  guix-build-d572dd305094/output/powerpc64-linux-gnu/bitcoin-d572dd305094-powerpc64-linux-gnu.tar.gz
b69af99e5df36f22469552763df81281a89c37be18ae095b18afa42e969dc8db  guix-build-d572dd305094/output/powerpc64le-linux-gnu/SHA256SUMS.part
1f89f6dc6e24163e87d6497c6254e74a45fa3e83dc763720b72e788007253a0c  guix-build-d572dd305094/output/powerpc64le-linux-gnu/bitcoin-d572dd305094-powerpc64le-linux-gnu-debug.tar.gz
4ec9e8b2ce4eddb6742e4abb1857cad5f1a497effee6772e4f5a63cb91d0a4f7  guix-build-d572dd305094/output/powerpc64le-linux-gnu/bitcoin-d572dd305094-powerpc64le-linux-gnu.tar.gz
d6db312e620494cb3b87240a3eebdccde60d23f6a023480f214e76353c4cd4fc  guix-build-d572dd305094/output/riscv64-linux-gnu/SHA256SUMS.part
a3a47e491cf1697dcf437bcdc9444f680c013bc09fc83c2f3c272645132ec954  guix-build-d572dd305094/output/riscv64-linux-gnu/bitcoin-d572dd305094-riscv64-linux-gnu-debug.tar.gz
de1eb124f65e42eb834a557e27b447d1f6c588a09a88e8a74ffba1fe52c01062  guix-build-d572dd305094/output/riscv64-linux-gnu/bitcoin-d572dd305094-riscv64-linux-gnu.tar.gz
7cfdec3057381bf70d1d4eecfdadd420544138ea2cf5be4978a5baac55b5879e  guix-build-d572dd305094/output/x86_64-apple-darwin19/SHA256SUMS.part
c4a053fdddbdcf73ae1978b5b8093faed505b49337ca659a452de8d1685a5369  guix-build-d572dd305094/output/x86_64-apple-darwin19/bitcoin-d572dd305094-osx-unsigned.dmg
957c7bc49189ec9be11d60440731d94f0e923c49df67763db4bff67369116ae5  guix-build-d572dd305094/output/x86_64-apple-darwin19/bitcoin-d572dd305094-osx-unsigned.tar.gz
261252243d0deaded0b0327fe5fad21b1c3f7686acbc47d1b3439a5db4a79773  guix-build-d572dd305094/output/x86_64-apple-darwin19/bitcoin-d572dd305094-osx64.tar.gz
2ed2512e7546ea2ecaa3c3d2808c93f660eb14bc938907301789f5b7a0c43d0a  guix-build-d572dd305094/output/x86_64-linux-gnu/SHA256SUMS.part
d8d1c715524174095bf480ea5d764817f7b274feea414419519e5950e1d55a1e  guix-build-d572dd305094/output/x86_64-linux-gnu/bitcoin-d572dd305094-x86_64-linux-gnu-debug.tar.gz
06e0ed8f685bc4bebbeb24265cd6aff1d8a8952e5d215ccfec527a34cd29e360  guix-build-d572dd305094/output/x86_64-linux-gnu/bitcoin-d572dd305094-x86_64-linux-gnu.tar.gz
7f2ac08986ab01e781998c7b1217abac0a548c1997475bbfbfccaed0d66ef04c  guix-build-d572dd305094/output/x86_64-w64-mingw32/SHA256SUMS.part
5571991b02b79fdb3aaa0debd50f3b31f8b68d55a09fb8317988c635d0ddf936  guix-build-d572dd305094/output/x86_64-w64-mingw32/bitcoin-d572dd305094-win-unsigned.tar.gz
904cf812dcc993e1f49d6c029ced0417821007c21e7c92bf1aaa30c957067fef  guix-build-d572dd305094/output/x86_64-w64-mingw32/bitcoin-d572dd305094-win64-debug.zip
b311715d5a702bed5bd0c661806257af80a56fb9770fa1a83049f0e512d7b0ab  guix-build-d572dd305094/output/x86_64-w64-mingw32/bitcoin-d572dd305094-win64-setup-unsigned.exe
cf46ce223059ec0893ceec6f5349a1ac04e250b49f88209574aa780bbff47c67  guix-build-d572dd305094/output/x86_64-w64-mingw32/bitcoin-d572dd305094-win64.zip
```

hebasto reviewed (approved):
ACK d572dd30509406289c591a8c842453113768e943, I verified updated Autoconf macros, I have reviewed the code and it looks OK, I agree it can be merged.

nit: spaces could be added between arguments of the `AC_DEFINE_UNQUOTED` macro in two places.

hebasto reviewed (commented):
After rebasing the following line seems need to be added to the 0ffcb115c062ab69cb82b67a397290e9a1687ccc commit:

https://github.com/bitcoin/bitcoin/blob/c1fb30633b6dcbf32db7d53c9f538019af80d6c5/configure.ac#L488

fanquake commented:
> need to be added to the 0ffcb11 commit:

Done.

hebasto reviewed (approved):
re-ACK 34094aff1348bd038ea2c5572c614631c7db3d6f

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
22097cd621cd88348f827b916f4b4b120b40c3515a3752595347e36d57dc9158  guix-build-34094aff1348/output/aarch64-linux-gnu/SHA256SUMS.part
43f10bb857afa7ea52a8ed9beed36ff0e3ee02dba31100fc04c8e0b2443d94eb  guix-build-34094aff1348/output/aarch64-linux-gnu/bitcoin-34094aff1348-aarch64-linux-gnu-debug.tar.gz
9095a8228376094065103f4adc9cdcb8189111fb9536ad88e4f8cecc3df6fc75  guix-build-34094aff1348/output/aarch64-linux-gnu/bitcoin-34094aff1348-aarch64-linux-gnu.tar.gz
9c73179059c6fe1f7643445ec5a530003fc41187aa0a94cb1f5c106097161e5b  guix-build-34094aff1348/output/arm-linux-gnueabihf/SHA256SUMS.part
70ddd4dd0a06c7491937084125f690c1d62fa2647c16048fc1a4a9a09d8b10b4  guix-build-34094aff1348/output/arm-linux-gnueabihf/bitcoin-34094aff1348-arm-linux-gnueabihf-debug.tar.gz
0a338fdc9788c33a0d519b6c09fdf6271e3bd68846ee61eef0a06a2df6bab419  guix-build-34094aff1348/output/arm-linux-gnueabihf/bitcoin-34094aff1348-arm-linux-gnueabihf.tar.gz
25eda7fae2984b9dadf47420d1dc09b0224d425144233482602bd9e6d348255c  guix-build-34094aff1348/output/dist-archive/bitcoin-34094aff1348.tar.gz
d70d84e43ffa2d809063cda868f708539e9114b2d14edb6ddcf05fdf73f3187b  guix-build-34094aff1348/output/powerpc64-linux-gnu/SHA256SUMS.part
8d1291e576d2b5f8f7120fe6e6ed4b23415249e22a657a350ccce68ff261e088  guix-build-34094aff1348/output/powerpc64-linux-gnu/bitcoin-34094aff1348-powerpc64-linux-gnu-debug.tar.gz
eab448186aee18ac33c39eed4d24501208d10d257fe6e2739adf589b1d4b693a  guix-build-34094aff1348/output/powerpc64-linux-gnu/bitcoin-34094aff1348-powerpc64-linux-gnu.tar.gz
4d28617b4d0ddb88c8b20d06ca21314ee40814043f92cabcd9ea3e3d8ee39183  guix-build-34094aff1348/output/powerpc64le-linux-gnu/SHA256SUMS.part
dea02168e170e92600012f5806ec8b39209282c2270669f2040682f74bc3f320  guix-build-34094aff1348/output/powerpc64le-linux-gnu/bitcoin-34094aff1348-powerpc64le-linux-gnu-debug.tar.gz
c7122e89d3186a183ac08e7f3020654722c98bf8acc8b790bb292b39f5ba8225  guix-build-34094aff1348/output/powerpc64le-linux-gnu/bitcoin-34094aff1348-powerpc64le-linux-gnu.tar.gz
0802a52720d2bec1264dc13f6554a9da347baa3d096242b29bb524f4b121eb10  guix-build-34094aff1348/output/riscv64-linux-gnu/SHA256SUMS.part
4aee9fc41f35d2adb2d0562902dd8584a8413a73c015ddcdcef00586779f63a7  guix-build-34094aff1348/output/riscv64-linux-gnu/bitcoin-34094aff1348-riscv64-linux-gnu-debug.tar.gz
3f0c4d6096ac7e08389e851c2d252632c044a700ce0174473ac4d7f66290e8cc  guix-build-34094aff1348/output/riscv64-linux-gnu/bitcoin-34094aff1348-riscv64-linux-gnu.tar.gz
c5149c46b9b7081d5715daf3e22fd30ffca23d333f664da1fabc8143ff8bf76c  guix-build-34094aff1348/output/x86_64-apple-darwin19/SHA256SUMS.part
81848355751e55a8a60636e3ea2f03ca6abb78736a5431715cd51cebd46bb961  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx-unsigned.dmg
d2f6f689cbedddd865f90dba9ddf21479c71c61b0350fda62804b2f233116a43  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx-unsigned.tar.gz
99d8fe428fcb67f9975e6b8d9a63d84946215a0a6b8f94967ce96cc3af4b7772  guix-build-34094aff1348/output/x86_64-apple-darwin19/bitcoin-34094aff1348-osx64.tar.gz
3dac13c7556d9a25ff5513bbb2638fe4fa74d8a88304bbdce52364df7832a3ab  guix-build-34094aff1348/output/x86_64-linux-gnu/SHA256SUMS.part
e4baa7da80fdabbb50953efaaa7b4867c7e575a7a156b728e8e197142df55697  guix-build-34094aff1348/output/x86_64-linux-gnu/bitcoin-34094aff1348-x86_64-linux-gnu-debug.tar.gz
f82f5bcc7197c1741b106f62be7b468aadbdf5b3198091582026cd450bf13b3a  guix-build-34094aff1348/output/x86_64-linux-gnu/bitcoin-34094aff1348-x86_64-linux-gnu.tar.gz
db22b5f48783917f985920ddb26aa170b4d6cc65112406548847a883099505b8  guix-build-34094aff1348/output/x86_64-w64-mingw32/SHA256SUMS.part
ead0809193ca1d462553a6f3f233cdbff7a3f8419100d825abfc10835508e485  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win-unsigned.tar.gz
e1006b6c114eaf33274144d8e9a20abc0cee01e26a4594063ee615bf09c1b344  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64-debug.zip
2d3fbc593b58d353a6859e02c0dd096d453cf5f1e3150c2a1c234bdfc97b4f24  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64-setup-unsigned.exe
25cfa8a3655727572593f100b7a70c1061fa6f3b017014ad7444059611c3ddda  guix-build-34094aff1348/output/x86_64-w64-mingw32/bitcoin-34094aff1348-win64.zip
```
